Mathlib/GroupTheory/SpecificGroups/Cyclic.lean

```diff
@@ -787,3 +787,106 @@ noncomputable def mulEquivOfPrimeCardEq {p : ℕ} [Fintype G] [Fintype H] [Group
   exact hG.trans hH.symm
 
 end ZMod
+
+section generator
+
+/-!
+### Groups with a given generator
+
+We state some results in terms of an explicitly given generator.
+The generating property is given as in `IsCyclic.exists_generator`.
+
+The main statements are about the existence and uniqueness of homomorphisms and isomorphisms
+specified by the image of the given generator.
+-/
+
+variable {G G' : Type*} [Group G] [Group G'] {g : G} (hg : ∀ x, x ∈ Subgroup.zpowers g) {g' : G'}
+
+@[to_additive]
+lemma orderOf_generator_eq_natCard : orderOf g = Nat.card G :=
+  Nat.card_zpowers g ▸ (Nat.card_congr <| Equiv.subtypeUnivEquiv hg)
```

Comment:
I think this should go below `orderOf_eq_card_of_forall_mem_zpowers` (or maybe pull that lemma here)

Reply:
And maybe other lemmas mentioning `∀ x : α, x ∈ zpowers a` should be here as well?

Reply:
OK; moving the lemma up (the other lemma is needed for proofs in between, so cannot be as easily moved).

Reply:
> And maybe other lemmas mentioning `∀ x : α, x ∈ zpowers a` should be here as well?

Which place is "here"? The bottom part of the file where I put the new stuff?

Reply:
Yes. I assume grouping similar lemmas boost visibility? But don't bother if it isn't possible.

Reply:
And I guess, based on the other lemmas using this condition, I should rename "generates" to "forall_mem_zpowers".

Reply:
Yeah I think so. Maybe we should define `IsGenerator x := ∀ x : α, x ∈ zpowers a` to enable dot products and better names some time in the future.

Reply:
> Yes. I assume grouping similar lemmas boost visibility? But don't bother if it isn't possible.

The other lemmas are distrbiuted all over the file, so grouping them would be a fair amount of work, which I think is a bit out of scope for this PR.

> Yeah I think so. Maybe we should define `IsGenerator x := ∀ x : α, x ∈ zpowers a` to enable dot products and better names some time in the future.

It certainly would make sense to either define such a predicate or a `Generator` structure  bundling an element with the predicate (or both?), but this would be a larger refactor of the material and needs more effort to be done right.

Reply:
Fair enough. This is still a great step forwards nonetheless.